SpatialGDK/Source/Private/Interop/Connection/SpatialWorkerConnection.cpp

```diff
@@ -124,18 +116,24 @@ void USpatialWorkerConnection::ConnectToLocator()
 	Worker_DeploymentListFuture_Get(DeploymentListFuture, nullptr, this,
 		[](void* UserData, const Worker_DeploymentList* DeploymentList)
 	{
+		USpatialWorkerConnection* SpatialConnection = static_cast<USpatialWorkerConnection*>(UserData);
+
 		if (DeploymentList->error != nullptr)
 		{
-			UE_LOG(LogSpatialWorkerConnection, Error, TEXT("Error fetching deployment list: %s"), UTF8_TO_TCHAR(DeploymentList->error));
+			const FString ErrorMessage = FString::Printf(TEXT("Error fetching deployment list: %s"), UTF8_TO_TCHAR(DeploymentList->error));
+			UE_LOG(LogSpatialWorkerConnection, Error, TEXT("Failed to connect to SpatialOS: %s"), *ErrorMessage);
+			SpatialConnection->OnConnectFailed.ExecuteIfBound(ErrorMessage);
+			return;
 		}
 
 		if (DeploymentList->deployment_count == 0)
 		{
-			UE_LOG(LogSpatialWorkerConnection, Error, TEXT("Received empty list of deployments. Failed to connect."));
+			const FString ErrorMessage = FString::Printf(TEXT("Received empty list of deployments."));
```

Comment:
@davedolben  Are the `FString::Printf`s required here?

Reply:
Oh lol of course not in this situation. Good catch. Will try to change as a drive-by fix in another PR soon